### PR TITLE
Ignore nonexistent block volumes in deletion of block-hosting volume.

### PR DESCRIPTION
**Pull request by obnoxxx**
### What does this PR achieve? Why do we need it?

Don't fail the deletion of a block-hosting volume if
it references a block volume id, that does not
exist in the database.

### CODE REVIEW

File: apps/glusterfs/app_volume.go [hunk 1]
```diff
@@ -252,17 +252,22 @@ func (a *App) VolumeDelete(w http.ResponseWriter, r *http.Request) {
...
+			_, err = NewBlockVolumeEntryFromId(tx, bvId)
+			if err == nil {
+				err = logger.LogError("Cannot delete a block hosting volume containing block volumes")
+				http.Error(w, err.Error(), http.StatusConflict)
+				return err
+			}
+			if err != ErrNotFound {
+				err = logger.LogError("Refusing to delete block-hosting volume: "+
+					"Error loading block-volume [%v]: %v", bvId, err)
+				http.Error(w, err.Error(), http.StatusConflict)
```
Comment (phlogistonjohn) [inline apps/glusterfs/app_volume.go]: Should this be conflict or just an internal server error?

File: apps/glusterfs/app_volume.go [hunk 2]
```diff
@@ -252,17 +252,22 @@ func (a *App) VolumeDelete(w http.ResponseWriter, r *http.Request) {
 			return nil
 		}
 
-		if volume.Info.BlockInfo.BlockVolumes == nil {
```
Comment (phlogistonjohn) [inline apps/glusterfs/app_volume.go]: Does the range below work as expected if the BlockVolumes value is nil? (I honestly don't remember)

File: apps/glusterfs/app_volume.go (hunk 1 above)
Comment (obnoxxx) [reply apps/glusterfs/app_volume.go]: You are probably right, thanks! Will change.

File: apps/glusterfs/app_volume.go (hunk 2 above)
Comment (obnoxxx) [reply apps/glusterfs/app_volume.go]: yes, I checked the following in the sandbox in golang.org ;-)
```
package main

import "fmt"

func main() {
	fmt.Println("Hello!")
	
	var mylist []string
	mylist = nil
	
	for _, myitem := range mylist {
		fmt.Println("myitem: %v", myitem)
	}
	
	fmt.Println("Bye...")
}
```

Comment (obnoxxx) [reply apps/glusterfs/app_volume.go]: it prints
```
Hello!
Bye...
```

Comment (phlogistonjohn) [reply apps/glusterfs/app_volume.go]: Great!

File: apps/glusterfs/app_volume.go [hunk 3]
```diff
@@ -252,17 +252,22 @@ func (a *App) VolumeDelete(w http.ResponseWriter, r *http.Request) {
...
+		for _, bvId := range volume.Info.BlockInfo.BlockVolumes {
+			_, err = NewBlockVolumeEntryFromId(tx, bvId)
+			if err == nil {
+				err = logger.LogError("Cannot delete a block hosting volume containing block volumes")
+				http.Error(w, err.Error(), http.StatusConflict)
+				return err
+			}
+			if err != ErrNotFound {
+				err = logger.LogError("Refusing to delete block-hosting volume: "+
+					"Error loading block-volume [%v]: %v", bvId, err)
```
Comment (obnoxxx) [inline apps/glusterfs/app_volume.go]: @phlogistonjohn one small doubt I had is whether this overriding err this way actually works?! ... :-)

Comment (phlogistonjohn) [reply apps/glusterfs/app_volume.go]: I'm not sure it is useful, but I think it will behave as you expect.

**Comment by obnoxxx** [comment]
@phlogistonjohn any additional thoughts? good to go?

**Pull request merged**